bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/ShellyHandlerFactory.java

```diff
@@ -0,0 +1,212 @@
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link ShellyHandlerFactory} is responsible for creating things and thing handlers.
+ *
+ * @author Markus Michels - Initial contribution
+ */
+@NonNullByDefault
+@Component(service = { ThingHandlerFactory.class,
+        ShellyHandlerFactory.class }, immediate = true, configurationPid = "binding.shelly")
+public class ShellyHandlerFactory extends BaseThingHandlerFactory {
+    private final Logger logger = LoggerFactory.getLogger(ShellyHandlerFactory.class);
+
+    @Nullable
+    private NetworkAddressService networkAddressService;
+    private final Set<ShellyDeviceListener> deviceListeners = new CopyOnWriteArraySet<>();
+
+    private static final Set<ThingTypeUID> SUPPORTED_THING_TYPES_UIDS = ShellyBindingConstants.SUPPORTED_THING_TYPES_UIDS;
+    private static boolean initialized = false;
+    private ShellyBindingConfiguration bindingConfig = new ShellyBindingConfiguration();
+    @Nullable
+    private ShellyCoapServer coapServer;
+
+    /**
+     * Activate the bundle: save properties
+     *
+     * @param componentContext
+     * @param configProperties set of properties from cfg (use same names as in
+     *            thing config)
+     */
+    @Activate
+    protected void activate(ComponentContext componentContext, Map<String, Object> configProperties) {
+        try {
+            super.activate(componentContext);
+
+            try {
+                if (!initialized) {
+                    Bundle bundle = this.getBundleContext().getBundle();
+                    Validate.notNull(bundle, "bundleContext must not be null!");
+                    Dictionary<String, String> d = bundle.getHeaders();
+                    Version v = bundle.getVersion();
+                    Validate.notNull(d, "bundleg.getHeaders() must not be null!");
+                    Validate.notNull(v, "bundleg.getVersion() must not be null!");
+                    @Nullable
+                    String name = d.get("Bundle-Name");
+                    Validate.notNull(name, "bundle name must not be empty!");
+                    String state = "";
+                    switch (bundle.getState()) {
+                        case Bundle.ACTIVE:
+                            state = "ACTIVE";
+                            break;
+                        case Bundle.INSTALLED:
+                            state = "INSTALLED";
+                            break;
+                        case Bundle.RESOLVED:
+                            state = "RESOLVED";
+                            break;
+                        default:
+                            state = new Integer(bundle.getState()).toString();
+                    }
+                    logger.info("{} Version {}.{}.{} (state={}, {}.jar, build {} UTC, location={})", name, v.getMajor(),
```

Comment:
Reduce logging to debug

Reply:
This allows to easily identify the installed version. Due to the fact that the user could get the jar from different sources (not just the dist package) I want to keep this info. 
Do we realy need to discuss 1 log time per bundle start?